### PR TITLE
Arbitrum snapshot proposals

### PR DESCRIPTION
**Pull request by ARDev097**
In this pull request, you'll find five SQL files, each catering to a distinct type of proposal available on the Arbitrum Snapshot platform. These files contain comprehensive data encompassing the various proposal types and the corresponding voting activities.

**1. governance_arbitrum_snapshot_single_choice_proposals:-** This file comprises data for single-choice type proposals, along with the associated voting details.

**2. governance_arbitrum_snapshot_basic_proposals:-** Here, you'll find data for basic type proposals, coupled with comprehensive voting records.

**3. governance_arbitrum_snapshot_approval_proposals:-** The contents of this file encompass data for approval type proposals and detailed voting records.

**4. governance_arbitrum_snapshot_ranked_choice_proposals:-** This file delves into data for ranked-choice type proposals, offering insights into the relevant voting information.
 
**5. governance_arbitrum_snapshot_weighted_proposals:-** This file includes data for weighted type proposals, presenting a comprehensive overview of the associated voting data.

### CODE REVIEW

**Comment by ARDev097** [comment]
Hello @Hosuke ,

Could you please take a moment to review and provide guidance on resolving this issue? Your assistance is highly valued, and any help you can offer would be greatly appreciated.

Thank you in advance!

image

File: models/governance/arbitrum/governance_arbitrum_snapshot_weighted_proposals.sql [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  dv.voter AS voter_address,
+  dv.vp AS voting_weightage,
+  dv.choice AS voting_choice,
+  dv.reason AS voting_reason,
+  FROM_UNIXTIME(dv.created) AS voting_timestamp,
+  YEAR(FROM_UNIXTIME(dv.created)) AS voting_year,
+  MONTH(FROM_UNIXTIME(dv.created)) AS voting_month,
+  DAY_OF_MONTH(FROM_UNIXTIME(dv.created)) AS voting_date
+FROM {{ source('dune.shot','dataset_proposals_view') }} AS dp
+JOIN {{ source('dune.shot','dataset_votes_view') }} AS dv
```
Comment (Hosuke) [inline models/governance/arbitrum/governance_arbitrum_snapshot_weighted_proposals.sql:30]: Hi @jeff-dude,
do you know how to use community uploaded data as source in the spell?
https://dune.com/queries/3490075?category=third_party_data&catalog=dune&schema=shot

Comment (Hosuke) [inline models/governance/arbitrum/governance_arbitrum_snapshot_weighted_proposals.sql:30]: Retry with other possible schema
```suggestion
FROM {{ source('dune','shot.dataset_proposals_view') }} AS dp
JOIN {{ source('dune','shot.dataset_votes_view') }} AS dv
```

File: sources/governance/arbitrum/governance_arbitrum_sources.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+version: 2
+
+sources: 
+  - name: dune.shot
+    description: >
+      Decoded table for proposals and their voting on Snapshot platform
+    tables:
+      - name: dataset_proposals_view
+      - name: dataset_votes_view
```
Comment (Hosuke) [inline sources/governance/arbitrum/governance_arbitrum_sources.yml:9]: Retry.
```suggestion
  - name: dune
    description: >
      Decoded table for proposals and their voting on Snapshot platform
    tables:
      - name: shot.dataset_proposals_view
      - name: shot.dataset_votes_view
```

File: models/governance/arbitrum/governance_arbitrum_snapshot_approval_proposals.sql [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    SPLIT(TRIM(TRIM(dp.choices, '[]'), ' '), ', '),
+    TRY_CAST(TRIM(TRIM(t.value, '[]'), ' ') AS INTEGER)
+  ) AS separated_voting_choice_name,
+  dv.reason AS voting_reason,
+  FROM_UNIXTIME(dv.created) AS voting_timestamp,
+  YEAR(FROM_UNIXTIME(dv.created)) AS voting_year,
+  MONTH(FROM_UNIXTIME(dv.created)) AS voting_month,
+  DAY_OF_MONTH(FROM_UNIXTIME(dv.created)) AS voting_date
+FROM {{ source('dune.shot','dataset_proposals_view') }} AS dp
+JOIN {{ source('dune.shot','dataset_votes_view') }} AS dv
```
Comment (Hosuke) [inline models/governance/arbitrum/governance_arbitrum_snapshot_approval_proposals.sql:35]: Hi @jeff-dude,
do you know how to use community uploaded data as source in the spellbook?
https://dune.com/queries/3490075?category=third_party_data&catalog=dune&schema=shot&id=01HR26V1S4TK4D0RNFZ6E4QCWD

Comment (ARDev097) [reply models/governance/arbitrum/governance_arbitrum_snapshot_approval_proposals.sql]: Hello @Hosuke ,
Thank you for the suggestion, but when using these schemas also I got the error.

image

Comment (Hosuke) [reply models/governance/arbitrum/governance_arbitrum_snapshot_approval_proposals.sql]: Sorry ser, my suggestion does not work.
We may need to find a new solution.

Comment (jeff-dude) [reply models/governance/arbitrum/governance_arbitrum_snapshot_approval_proposals.sql]: this is now unblocked, you can pass a `database` arg to the source jinja and override the default with `dune` or whatever is needed
for an example, please see `tokens_erc20.sql`

**Comment by jeff-dude** [comment]
closing due to inactivity

**Pull request closed without merging**